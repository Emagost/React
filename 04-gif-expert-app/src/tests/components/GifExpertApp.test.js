import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../../components/GifExpertApp';

describe('Test GifExpertApp', () => {
    
    test('snapshot', () => {
        
        const wrapper = shallow( <GifExpertApp /> );
        expect(wrapper).toMatchSnapshot();

    });

    test('should return a list of categories', () => {

        const categories = ['The ofice', 'Parasyte'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );
        
        expect(wrapper).toMatchSnapshot();
        // expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
    })
    
    


})
