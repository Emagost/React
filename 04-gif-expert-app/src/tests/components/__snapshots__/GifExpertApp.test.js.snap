// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifExpertApp should return a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category=""
      key=""
    />
  </ol>
</Fragment>
`;

exports[`Test GifExpertApp snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category=""
      key=""
    />
  </ol>
</Fragment>
`;
